spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/profile/AbstractKubernetesProfileEnvironmentPostProcessor.java

```diff
@@ -58,12 +58,17 @@ public void postProcessEnvironment(ConfigurableEnvironment environment, SpringAp
 
 		application.addInitializers(ctx -> LOG.replayTo(AbstractKubernetesProfileEnvironmentPostProcessor.class));
 
-		boolean kubernetesEnabled = environment.getProperty("spring.cloud.kubernetes.enabled", Boolean.class, true);
-		if (!kubernetesEnabled) {
-			return;
+		boolean serviceHost = StringUtils
```

Comment:
this is another important change, we need to replicate what `@ConditionalOnCloudPlatform(KUBERNETES)` is doing

Reply:
Why?

Reply:
this is an environment post processor here. We used to have : 

```
boolean kubernetesEnabled = environment.getProperty("spring.cloud.kubernetes.enabled", Boolean.class, true);
```

because there is simply no other way to check if kubernetes was enabled or not, as this is an environment post processor declared in `spring.factories`. 

We need to have the same logic here, but for `@ConditionalOnCloudPlatform(KUBERNETES)`.

Reply:
What if you check spring.main.platform?

Reply:
what if users don't specify that, but they are still running under kubernetes? meaning `KUBERNETES_SERVICE_HOST` and `KUBERNETES_SERVICE_PORT` are present? Wouldn't we miss adding `kubernetes` profile?

Reply:
I am assuming Boot would set the value correctly based on the platform, is that not the case?

Reply:
does not look like it, see [here](https://github.com/spring-projects/spring-boot/blob/main/spring-boot-project/spring-boot/src/main/java/org/springframework/boot/cloud/CloudPlatform.java#L161..L164).

Reply:
Could we use the CloudPlatform class instead or replicating the logic?

Reply:
OMG! I am an idiot, thx a lot.

Reply:
No worries :)